### DISCUSSION TITLE
tui-select disabled

### DISCUSSION THREAD

**Discussion by expdts** [taiga-family/taiga-ui#6295] [Q&A] [answered]
Подскажите пожалуйста, как я могу выделить цветом и заблокировать выбор в `tui-select` используя `tui-data-list-wrapper`?
Как ожидается, блокируем выбор и выделяем цветом элемент имеющий `isActive = false`:
![image](https://github.com/taiga-family/taiga-ui/assets/109337333/c2ac0105-33a4-4d55-88c4-3cd44ad1f57c)

[Пример](https://stackblitz.com/edit/angular-tmuv65?file=src%2Fapp%2Fapp.component.ts)

**Comment by waterplea** [comment]
Для более точечного контроля используй `tui-data-list`. Блокировать выбор в `tui-data-list-wrapper` можно через инпут `disabledItemHandler`

**Comment by expdts** [comment]
@waterplea можете показать на моем примере как правильно использовать в моем случае `tui-data-list` используя `formControlName`? У меня не получилось.

Не очень понял функционал `disabledItemHandler`, думал использовать его как функцию. Например так `disabledItemHandler="changeData(item)"` но не понятно как передать в него объект `item`, для возврата `true` или `false`.

Предполагается использование сложной функции для блокирования и выделения текста цветом. Например может быть такое условие блокирования и выделения цветом: `!item.isActive && item.pl.codeId === 2`

**Reply by waterplea** [reply, marked as answer]
В твоём примере вот такая функция подойдёт:
```ts
readonly handler = item => !item.isActive
```
```html
<tui-data-list-wrapper
  *tuiDataList
  [items]="listData"
  [itemContent]="testListDataContent"
  [disabledItemHandler]="handler"
/>
```
Если использовать `tui-data-list`, то там все `tuiOption` будут в шаблоне и можно написать:
```html
<button *ngFor="..." tuiOption [disabled]="!item.isActive" ...
```
А в стилях прописать:
```css
[tuiOption]:disabled {
  background: pink;
}
```

**Reply by expdts** [reply]
Спасибо за пример `disabledItemHandler`, изначально думал что необходимо всегда использовать `TuiBooleanHandler`.
Больше всего для моего случая мне подошел `tui-data-list`. Использование стиля `[tuiOption]:disabled` мне не подходит. Реализовал через установку стиля в `<button ... [style]="{'background': !item.isActive ? 'pink' : ''}"`, это позволяет сделать только для определенного.
